### PR TITLE
finish question 4

### PR DESCRIPTION
**Pull request by MattyB2112**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,4 +47,40 @@ describe("/api", () => {
...
+          expect(body.message).toBe("not found");
+        });
+    });
+    test("GET /api/articles/100 returns a 200 and empty object as parameter is valid but doesn't exist in database", () => {
+      return request(app)
+        .get("/api/articles/100")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body).toEqual({});
+        });
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:83]: Nice thinking on including some error handling here, however lets rethink the situations in the last two tests.
If I ask for article 100, and it doesn't exist, this should be a 'not found' error and relevant code.
If I ask for article 'banana', this isn't a valid request and should be communicated as such. Have a think about which two codes we'd need in these situations and allow your error handlers to pick these up 👍

File: models/topics.model.js [hunk 2]
```diff
@@ -13,4 +13,14 @@ exports.fetchEndpoints = () => {
   });
 };
 
-// endpoints.json
+exports.fetchArticleById = (id) => {
+  const paramAsNum = Number(id);
+  if (Number.isNaN(paramAsNum)) {
```
Comment (nc-richard-neat) [inline models/topics.model.js:18]: Lets allow psql to pick up the error if it is going to, and only if it doesn't throw an error do we need to consider a custom error here.

**Review by nc-richard-neat — COMMENTED** [review]
Just a couple of suggestions below to improve the error handling :)

File: __tests__/app.test.js (hunk 1 above)
Comment (MattyB2112) [reply __tests__/app.test.js:83]: Hi Rich! Thank you for the feedback. Normally I would have allowed PSQL to pick up the error and logged 100 as a 404, but after the lecture today with Liam I was left a bit confused, as he said that technically 100 is correct, but its just not in the database yet but could be at some point, so returning an empty object would be the right thing to do? On all previous sprints I've used psql so happy to change back to this!

Comment (nc-richard-neat) [reply __tests__/app.test.js:83]: PSQL won't pick up 100, but it should pick up 'banana'. For 100 you will simply get back an empty array, so if we get an empty array back we know that means the article hasn't been found and can handle it accordingly :)

**Pull request merged**